t/re/reg_mesg.t

```diff
@@ -20,6 +20,36 @@ use open qw(:utf8 :std);
 use warnings;
 BEGIN { ${^WARNING_BITS} = undef }  # Kludge to restore default warnings
 
+# Escape a string in a similar fashion to how the regex debugger does,
+# using "x" style escapes in the form "%x{..}" to show the codepoint of
+# the escaped value and "%%" to show a literal percentage sign.  We use
+# percentage instead of backslash so that it is trivial to distinguish
+# backslash based sequences that are in the pattern from escaped octets.
+# Currently we assume ASCII/Unicode semantics, and escape anything that
+# is outside of codepoint 32 to codepoint 126, with the exception of "%"
+# itself.  This is only used for the test names to make it easier to
+# figure out which test case is failing.  Note that we deliberately avoid
+# using the regex engine to do the escaping. - Yves
+
+sub escape_string {
+    my ($str) = @_;
+    my $escaped = "";
+    for my $i (0 .. length($str)-1) {
+        my $char = substr($str,$i,1);
+        my $ord = ord($char);
+        if ($char eq "%") {
+            $escaped .= "%%";
+        }
+        elsif (32 <= $ord <= 126) {
```

Comment:
This won't work on non-ASCII platforms.  Instead write 
`elsif (32 <= utf8::native_to_uni($ord) <= 126) {`
The extra function call is optimized out on ASCII platforms.

Reply:
Thanks. Done. BTW, the function is utf8::native_to_unicode().  I believe the C version is "uni", but in the perl level we spell it out.